### ISSUE TITLE
PinnedDrop implementations can contain unsafe code without an unsafe block

### ISSUE THREAD

**Issue by sfackler** [taiki-e/pin-project#148]
Since the `#[pinned_drop]` attribute injects an `unsafe` onto the drop fn, code within the drop implementation can perform unsafe operations without an explicit unsafe block.

You could potentially instead generate something like

```rust
impl PinnedDrop for Foo {
    unsafe fn drop(self: Pin<&mut Self>) {
        fn drop_inner(self: Pin<&mut Self>) {
            // user code here
        }
        drop_inner(self);
    }
}
```

**Comment by Aaron1011** [comment]
This is what [my original PR did](https://github.com/taiki-e/pin-project/pull/18/files#diff-378c6adb76da4c8aef9b9b0ebd093aabR157-R172). It looks like this got changed at some point.

**Comment by taiki-e** [comment]
@sfackler Thanks for finding this! Seems I accidentally [changed this](https://github.com/taiki-e/pin-project/pull/86/files#diff-49477733152d46aa097301cc9cecf5ddL85-R90) in #86...
I will fix this soon.

**Comment by taiki-e** [comment]
Published 0.4.4 which fixes this issue.

**Comment by sfackler** [comment]
Thanks!
